### PR TITLE
[android] Fix unchecked ops and deprecations

### PR DESCRIPTION
**Pull request by cco3**
Additionally, add checks for these things to build.gradle.

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java [hunk 1]
```diff
@@ -183,7 +183,7 @@ public int onStartCommand(Intent intent, int flags, int startId) {
     initializeLists();
     // Start scanning only if the screen is on
     PowerManager powerManager = (PowerManager) getSystemService(Context.POWER_SERVICE);
-    if (powerManager.isScreenOn()) {
+    if (powerManager.isInteractive()) {
```
Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: isInteractive is not available in API 19 which is our min SDK

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: good catch

**Comment by cco3** [comment]
done

**Comment by g-ortuno** [comment]
LGTM

**Pull request merged**